src/QirRuntime/public/QubitAllocationPolicies.hpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <stack>
+
+#include "BitStates.hpp"
+#include "IIdAllocationPolicy.hpp"
+
+using namespace std;
+namespace Microsoft
+{
+namespace Quantum
+{
+    /*==============================================================================
+        The default qubit manager reuses released qubit ids in "Last released -
+        first reused" order.
+    ==============================================================================*/
+    class CReuseLastReleasedQubitAllocationPolicy : public IIdAllocationPolicy

```

Comment:
I assume there could be other implementations of this interface, so it's not clear to me why these two are considered part of the public API.

In general, I would expect a folder/project that held the header files for types that any code that uses a simulator or execution engine would need to know about, and a separate folder/project that defines the types that an implementation of a simulator or execution engine needs to have access to. I would expect actual implementations to be in yet other folders, either one per folder or perhaps a few lumped together if they're small. The current "public" folder seems to be a mix of external types, internal types, and specific implementation types.